### PR TITLE
Release v0.0.9-BETA (PM 5.0.0)

### PR DESCRIPTION
**Pull request by xRodzMC**
**This is an updated version of dev-commit. These needs to be tested in PocketMine-MP 5.0.0 in the future.**

Dylan (dktapps) has announcement about the PocketMine-MP 5.0.0 which supports many advantages of 4.0.0, this release will be able to use to test them to 5.0.0. When its done and ready to publish, we will publish the v0.2.0 which supports 5.0.0 version of PocketMine-MP.

*If you have a question about this pull request, you can comment down below or review this commit.*

### CODE REVIEW

**Comment by xRodzMC** [comment]
Re-updated branch.

**Review by NhanAZ — DISMISSED** [review]

File: build_info.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
   "plugin_name": "ProfanityFilter",
-  "version": "0.0.8",
+  "version": "0.0.9",
```
Comment (Reinfy) [inline build_info.json]: This should be now `v0.1.0-BETA`

File: build_info.json [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
   "channel": "stable",
```
Comment (Reinfy) [inline build_info.json]: And the channel must be `BETA`

File: build_info.json [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
-  "api": "4.15.2",
+  "api_from": "4.0.0",
```
Comment (Reinfy) [inline build_info.json]: This should be `4.12.1`

File: build_info.json [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
+  "api_to": "4.15.2",
```
Comment (Reinfy) [inline build_info.json]: This should be `5.0.0`

File: src/ReinfyTeam/ProfanityFilter/Loader.php [hunk 5]
```diff
@@ -107,7 +107,12 @@ private function registerCommands() : void {
 	}
 
 	private function checkUpdate() : void {
-		$this->getServer()->getAsyncPool()->submitTask(new PoggitUpdateTask($this->getDescription()->getName(), $this->getDescription()->getVersion()));
+		$lang = new Language();
+		if ((bool) ($this->getConfig()->get("check-updates") ?? true)) {
```
Comment (Reinfy) [inline src/ReinfyTeam/ProfanityFilter/Loader.php]: Why you are checking of a non-nullable function? The function will return false when it is null.

**Review by Reinfy — CHANGES_REQUESTED** [review]

**Comment by Reinfy** [comment]
Too many code issue stability must be resolved before going to merge this pull request.

File: src/ReinfyTeam/ProfanityFilter/Loader.php [hunk 6]
```diff
@@ -108,7 +108,7 @@ private function registerCommands() : void {
 
 	private function checkUpdate() : void {
 		$lang = new Language();
-		if ((bool) ($this->getConfig()->get("check-updates") ?? true)) {
+		if (($this->getConfig()->get("check-updates")) {
```
Comment (Reinfy) [inline src/ReinfyTeam/ProfanityFilter/Loader.php]:
```suggestion
		if ($this->getConfig()->get("check-updates")) {
```

**Review by Reinfy — CHANGES_REQUESTED** [review]

**Review by Reinfy — APPROVED** [review]

**Pull request merged**